applications/ALEapplication/custom_elements/laplacian_meshmoving_element.cpp

```diff
@@ -55,179 +55,227 @@
 
-void LaplacianMeshMovingElement::EquationIdVector(
-    EquationIdVectorType& rResult, ProcessInfo& rCurrentProcessInfo) {
-
-  GeometryType& rGeom = this->GetGeometry();
-  const SizeType NumNodes = rGeom.size();
-  unsigned int dimension = GetGeometry().WorkingSpaceDimension();
-  const SizeType LocalSize = NumNodes * dimension;
-
-  if (rResult.size() != LocalSize)
-    rResult.resize(LocalSize, false);
-
-  for (SizeType iNode = 0; iNode < NumNodes; ++iNode) {
-
-    SizeType Index = iNode * dimension;
-    rResult[Index] = rGeom[iNode].GetDof(MESH_DISPLACEMENT_X).EquationId();
-    rResult[Index + 1] = rGeom[iNode].GetDof(MESH_DISPLACEMENT_Y).EquationId();
-    if (dimension == 3)
-      rResult[Index + 2] = rGeom[iNode].GetDof(MESH_DISPLACEMENT_Z).EquationId();
-  }
+void LaplacianMeshMovingElement::EquationIdVector(EquationIdVectorType& rResult,
+                                                  ProcessInfo& rCurrentProcessInfo)
+{
+    GeometryType& rGeom = this->GetGeometry();
+    const SizeType NumNodes = rGeom.size();
+    unsigned int dimension = GetGeometry().WorkingSpaceDimension();
+    const SizeType LocalSize = NumNodes * dimension;
+
+    if (rResult.size() != LocalSize)
+        rResult.resize(LocalSize, false);
+
+    for (SizeType iNode = 0; iNode < NumNodes; ++iNode)
+    {
+        SizeType Index = iNode * dimension;
+        rResult[Index] = rGeom[iNode].GetDof(MESH_DISPLACEMENT_X).EquationId();
+        rResult[Index + 1] = rGeom[iNode].GetDof(MESH_DISPLACEMENT_Y).EquationId();
+        if (dimension == 3)
+            rResult[Index + 2] = rGeom[iNode].GetDof(MESH_DISPLACEMENT_Z).EquationId();
+    }
 }
 
 void LaplacianMeshMovingElement::GetDofList(DofsVectorType& rElementalDofList,
-                                            ProcessInfo& rCurrentProcessInfo) {
-
-  GeometryType& rGeom = this->GetGeometry();
-  const SizeType NumNodes = rGeom.size();
-  unsigned int dimension = GetGeometry().WorkingSpaceDimension();
-  const SizeType LocalSize = NumNodes * dimension;
-
-  if (rElementalDofList.size() != LocalSize)
-    rElementalDofList.resize(LocalSize);
-
-  for (SizeType iNode = 0; iNode < NumNodes; ++iNode) {
-
-    SizeType Index = iNode * dimension;
-
-    rElementalDofList[Index] = rGeom[iNode].pGetDof(MESH_DISPLACEMENT_X);
-    rElementalDofList[Index + 1] = rGeom[iNode].pGetDof(MESH_DISPLACEMENT_Y);
-    if (dimension == 3)
-      rElementalDofList[Index + 2] = rGeom[iNode].pGetDof(MESH_DISPLACEMENT_Z);
-
-  }
+                                            ProcessInfo& rCurrentProcessInfo)
+{
+    GeometryType& rGeom = this->GetGeometry();
+    const SizeType NumNodes = rGeom.size();
+    unsigned int dimension = GetGeometry().WorkingSpaceDimension();
+    const SizeType LocalSize = NumNodes * dimension;
+
+    if (rElementalDofList.size() != LocalSize)
+        rElementalDofList.resize(LocalSize);
+
+    for (SizeType iNode = 0; iNode < NumNodes; ++iNode)
+    {
+        SizeType Index = iNode * dimension;
+
```

Comment:
getting dofs key is a little expensive. you can get some optimization by doing the following

    unsigned int pos = this->GetGeometry()[0].GetDofPosition(MESH_DISPLACEMENT_X);
     for (SizeType iNode = 0; iNode < NumNodes; ++iNode)
      {
        SizeType Index = iNode * dimension;
          rElementalDofList[Index] = rGeom[iNode].pGetDof(MESH_DISPLACEMENT_X,pos);
          rElementalDofList[Index + 1] = rGeom[iNode].pGetDof(MESH_DISPLACEMENT_Y,pos+1);
          if (dimension == 3) //OF COURSE THIS SHOULD GO OUT OF THE LOOP
            rElementalDofList[Index + 2] = rGeom[iNode].pGetDof(MESH_DISPLACEMENT_Z,pos+2);